src/Components/Steps/Steps.tsx

```diff
@@ -0,0 +1,141 @@
+    const handleMediaUpload = async (
+      event: React.ChangeEvent<HTMLInputElement>
+    ) => {
+      if (event.target.files === null) return;
+      const files: FileList = event.target.files;
+      Array.from(files).forEach((file: File) => {
+        if (!mediaPresenter.addFile(file)) {
+          showError(file.name + "is not an acceptable file format");
+        }
+      });
+    };
+
+    const [stepOneOpen, setStepOneOpen] = useState<boolean>(true);
+    const [stepTwoOpen, setStepTwoOpen] = useState<boolean>(false);
+
+    return (
+      <div className={styles.stepsContainer}>
+        <div
+          className={styles.steppingContainer}
+          onClick={() => {
+            if (mediaPresenter.mediaReady) {
+              if(!stepOneOpen) {
+                setStepOneOpen(true);
+              }
+              closePlayerModal();
+            }         
+          }}
+        >
+          <span className={styles.stepNumber}>1</span>
+
+          <p className={styles.stepText}>Add your files</p>
+          <span onClick={()=> {
+            setStepOneOpen(!stepOneOpen);
+          }}>
+            <i className={stepOneOpen ? styles.downArrow : styles.upArrow}></i>
+          </span>
+        </div>
+
+        {stepOneOpen && (
+          <button className={styles.uploadBtnCont}>
+            <label htmlFor="fileUpload">
+              <div className={mediaPresenter.mediaReady ? styles.clickedUploadBtn : styles.uploadBtn }>Upload media</div>
+            </label>
+            <input
+              hidden
+              multiple
+              id="fileUpload"
+              type="file"
+              accept="video/* image/*"
+              onChange={handleMediaUpload}
+            />
+          </button>
+        )}
+          { stepOneOpen &&
+            mediaPresenter.mediaReady &&
+        <button className={styles.continueBtn}
+            onClick={()=> {
+              if(!stepTwoOpen) {
+                setStepTwoOpen(true);
+                setStepOneOpen(false);
+                openPlayerModal();
+              }
+            }}
+          >Continue</button>
+          }
+        <div
+          className={
+            mediaPresenter.mediaReady
+              ? styles.steppingContainer
+              : styles.disbContainer
+          }
+          onClick={() => {
+            if (mediaPresenter.mediaReady) {
+              if(!stepTwoOpen) {
+                setStepTwoOpen(true);
+              }
+                openPlayerModal();
+            }
+        }}
+        >
+          <div
+            className={
+              mediaPresenter.mediaReady
+                ? undefined
+                : styles.disbSteppingContainer
+            }
+          >
+            <span
+              className={
+                mediaPresenter.mediaReady
+                  ? styles.stepNumber
+                  : styles.disbStepNumber
+              }
+            >
+              2
```

Comment:
Looks like this component could be broken down into a StepsContainer and children `Step` components

Reply:
Yep I can do that.. is it because the codes is too long or is it because it looks like there is repetitive code?

If it's the latter, each component has a different css style and calls different functions to get different information in the mediapresenter and outputPresenter. So I'm unable to compress them with Objects.map or arrays sort of approach.

Reply:
So if I were to break it down into StepsContainer, Step1 and Step2, it will be very similar codes but on different files. Is that preferred?

Reply:
@liviaw 

The benefit of breaking these down into smaller components is that the code becomes easier to reason about. You can then skim through the component and get a pretty good feel for what is going on without caring too much about the actual rendered html.

For example, if you were break step two into its own component, the entry in the `Steps` component could become
```ts
<StepTwo
    isOpen={stepTwoOpen}
    closeStep={() => setStepTwoOpen(false) }
    openStep={() => setStepTwoOpen(true)}
    isMediaReady={mediaPresenter.mediaReady}
    openPlayer={openPlayerModal}
  />
```

where the `StepTwo` component might look like:

```ts
const StepTwo = ({
  isOpen,
  isMediaReady,
  openStep,
  closeStep,
  openPlayer,
}: {
  isOpen: boolean,
  isMediaReady: boolean,
  openStep(): void,
  closeStep(): void,
  openPlayer(): void,
}) => {
  const onOpenStep = useCallback(() => {
    if (isMediaReady) {
      if (!isOpen) {
        openStep()
      }
      openPlayer();
    }
  }, [isOpen, isMediaReady, openStep, openPlayer])

  const onArrowClick = useCallback(() => {
    isOpen ? closeStep() : openStep()
  }, [isOpen, closeStep, openStep])

  return (
    <div
      className={isMediaReady ? styles.steppingContainer : styles.disbContainer}
      onClick={onOpenStep}
    >
      <div
        className={isMediaReady ? undefined : styles.disbSteppingContainer
        }
      >
        <span className={isMediaReady ? styles.stepNumber : styles.disbStepNumber}>
          2
        </span>
        <p className={isMediaReady ? styles.stepText : styles.disbStepText}>
          Customise your video
        </p>
        <span>
          <i
            className={
              isMediaReady
                ? isOpen ? styles.downArrow : styles.upArrow
                : styles.disabledUpArrow
            }
            onClick={onArrowClick}
          ></i>
        </span>
      </div>
    </div>
  )
}
```

Side note: As long as the separate component isn't too big, it can still live within the same file.